### ISSUE TITLE
False positive for schema validation when a variable is used

### ISSUE THREAD

**Issue by ribeaud** [ansible/ansible-lint#2175]
##### Summary

[this](https://raw.githubusercontent.com/ansible-community/schemas/main/f/ansible.json#/$defs/tasks) schema specifies that **Retries** should be of type `integer`. In my case **Retries** is retrieved from a variable: `retries: "{{service_container_pull_retry_count }}"`. I have to use the quotes because it is a variable.

I've found a similar issue on [stackoverflow.com](https://stackoverflow.com/questions/28234430/ansible-integer-variables-in-yaml).

##### Issue Type

This warning is a false positive but I did not find a way how I could skip it by reading the [documentation](https://ansible-lint.readthedocs.io/en/latest/rules/#false-positives-skipping-rules).

##### Ansible and Ansible Lint details

```ansible-lint -w "" -x yaml
WARNING  Listing 1 violation(s) that are fatal
schema: '{{ service_container_pull_retry_count }}' is not of type 'integer' (schema[tasks])
roles/service/tasks/main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``playbook-schema``,
``tasks-schema``.

This rule is not skippable and stops further processing of the file.

Schema bugs should be reported towards (https://github.com/ansible/schemas) project instead of ansible-lint.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
  - schema  # Perform JSON Schema Validation for known lintable kinds.

Finished with 1 failure(s), 0 warning(s) on 39 files.
```

##### OS / ENVIRONMENT

**macos**, `ansible-lint` installed via **Python**.

##### Desired Behavior

Either we should relax the rule in such cases, or we should be able to skip rule validation for this specific line.

##### Actual Behavior

I am getting a failure.

**Comment by ribeaud** [comment]
Thank you! That was quick.

A dump question from my side: how can I profit from the change? Is this a new release of `ansible-lint`? A new release of `ansible`? Or should it already be fixed as soon as I run `ansible-lint -w "" -x yaml`?

**Comment by ribeaud** [comment]
@ssbarnea: might I ask you when I would be able to run my command above without getting any failure? Or what should I do to profit from the fixes? Should I wait until the next release (if so, when would **ansible-lint** be released)?

**Comment by MallocArray** [comment]
@ssbarnea  I'm running 6.2.2 and still getting the same error for a retries value. I'm not using a variable, but using the random option.  Same issue as this, or should I open a new Issue?

```yaml
      retries: 10
      delay: "{{ 30 | random }}"
```

```text
WARNING  Listing 1 violation(s) that are fatal
schema: '{{ 30 | random }}' is not of type 'integer' (schema[playbook])
```

```bash
ansible-lint --version
ansible-lint 6.2.2 using ansible 2.12.6
```
